### DISCUSSION TITLE
Will Malt support 4.2 eventually?

### DISCUSSION THREAD

**Discussion by Dereuz24** [bnpr/Malt#569] [General]
Hi there  !

I was wondering if  @pragma37 had the intention to eventually release a Malt version for 4.2 in the future? 

I understand you are currently working on the EEVEE NPR prototype but that will take some time and a lot of work to come together, but as long as Malt is only for 4.1 starting to migrate to 4.2 is completely out of the question for some of us after seeing what this engine can do

EEVEE legacy had some limitations but going from Malt to EVEE NEXT for NPR..

**Comment by aVersionOfReality** [comment]
Yeah, Eevee next is causing a lot of problems. Malt more needed than ever!

**Reply by Dereuz24** [reply]
Absolutely

"He who has a MALT and a compositor to work with can bear almost any EEVEE NEXT" ― Friedrich Nietzsche

**Reply by pragma37** [reply]
> Eevee next is causing a lot of problems.

😥

**Comment by aVersionOfReality** [comment]
It would also be great if Malt was made available through the new Blender Extensions. I'm not sure what's involved with that though.

**Reply by pragma37** [reply]
I'm a bit out of the loop with that.
I think initially addons were supposed to be reviewed for approval, but I don't think that's the case anymore?

We also have the problem of Malt being tied to a specific Blender version, but that should be solved soon.
However, Malt still has different versions for each OS and doesn't even work on Mac, so I'm not sure if it fits the platform intent.

**Comment by pragma37** [comment]
Sure! It's just that sometimes it takes me a few days to make the update.

The good news is that, starting from 4.2, Malt won't be tied to a specific Blender version.

The reason for this was due to some workarounds that I had to implement for extracting the Blender mesh data in a performant way, but most of those workarounds are no longer necessary, and the only required workaround left was for extracting mesh tangents.
However, I'm preparing a patch to compute mesh tangents manually instead of extracting them from Blender, so except for Python API breaks, it should be possible to use any Malt version with any Blender version.

**Reply by Dereuz24** [reply]
That's amazing news! Having to manually set Malt up for each version had me worried about something breaking in the future but this is just way better than I expected. Thanks a lot!

I have another question, I remember reading @aVersionOfReality thread on using the compositor for multiple scenes while using EEVEE on one and Malt on another one:

https://github.com/bnpr/Malt/discussions/199

I did try that for a small project I did recently and it works amazingly well, but it's true that once Malt runs and you go back to the EEVEE scene and then back to Malt, it just doesn't keep rendering unless you manually reinitialize it. Do you think it could be possible to make the render engine reinitialize automatically to avoid this issue?

I make only special emphasis on this because for any feature that's already a solved problem on EEVEE or Cycles we could just use that setup/workflow while still using Malt and not having to depend on specific functionality or remaking nodes that we already have on EEVEE and keep the best things of both worlds

But there's still some sort of issue/conflict when using MALT in different scenes with different engines

**Comment by pragma37** [comment]
I forgot to comment, but the new release is ready. 🙂

**Reply by Dereuz24** [reply]
Thanks a lot, it works perfectly!
